File: 75-sort-colors/sort-colors.cpp
class Solution
{
public:
    void sortColors(vector<int> &nums)
    {
        int ones = 0, tows = 0, zeros = 0;
        for (auto it : nums)
        {
            if (it == 0)
                zeros++;
            if (it == 1)
                ones++;
            if (it == 2)
                tows++;
        }
        int i = 0;
        for (; i < zeros; i++)
            nums[i] = 0;
        for (; i < zeros + ones; i++)
            nums[i] = 1;
        for (; i < zeros + ones + tows; i++)
            nums[i] = 2;
    }
};

File: 75-sort-colors/sort-colors.java
class Solution {
    public void sortColors(int[] nums) {
        int s = 0;
        int curr = 0;
        int e = nums.length - 1;
        while(curr <= e)
        {
            if(nums[curr] == 0)
            {
                nums[curr] = (nums[curr] + nums[s]) - (nums[s] = nums[curr]);
                s++;
                curr++;
            }
            else if(nums[curr] == 2)
            {
                nums[curr] = (nums[curr] + nums[e]) - (nums[e] = nums[curr]);
                e--;
                if(nums[curr] == 1)
                    curr++;
            }
            else
                curr++;
        }
    }
}